### DISCUSSION TITLE
How to make Radeon RX Vega M / GH work?

### DISCUSSION THREAD

**Discussion by k4xx0r** [osy86/HaC-Mini#972] [Q&A]
Hello,

I would like to ask if it's possible to provide us with some more info on how you got the Radeon RX Vega M or Vega GH to work. 

My intention is to use KVM to run virtualized Hackintoshes on a Hades Canyon with GPU passthrough to the macOS guests. I successfully created a macOS VM with [OSX-KVM](https://github.com/kholia/OSX-KVM) and can passthrough the PCI device, but I never get any video output.

I tried to spoof the device-id just like you did in [SSDT-GpuSpoof.asl](https://github.com/osy/HaC-Mini/blob/c55ab62198661b1ccd5dbaef19803c15cae6954d/ACPI/SSDT-GpuSpoof.asl#L51) by telling QEMU to spoof it like this (I have no idea how to do it with the asl as I can't find the proper value for Scope):
`    <qemu:arg value="-set"/>
    <qemu:arg value="device.hostdev0.x-pci-device-id=26591"/>`
The effect is that the GPU gets recognized as RX480 and doesn't claim that no kext was loaded, but still no display output. I also included your Polaris22Fixup.kext to my OpenCore but that also never changed anything. 

Just to verify whether my vfio passthrough configuration is correct I tried to passthrough to a Linux EFI-based VM and there I get a proper accelerated display output, so I suppose there's something in the Hackintosh to be fixed.

Any ideas? 

Anyways, I wanted to also thank you for your hard work so far. This project is really nice and you can be proud of such nice craftsmanship.

**Comment by osy** [comment]
You can’t just spoof it at that level as you’ll not get the right driver code. It has to be spoofed the same way to get the drivers to match but internally when it queries the device it’ll still get the Vega M code.

**Reply by k4xx0r** [reply]
I see. Can you maybe help me then to adjust [SSDT-GpuSpoof.asl](https://github.com/osy/HaC-Mini/blob/master/ACPI/SSDT-GpuSpoof.asl#L51)?

A Windows disk running inside the Hackintosh VM reports this as location path for the passed through Vega GH, but I have no idea how to translate this into a proper ACPI path, as every combination I tried yields a syntax error for the ASL compiler:

`ACPI(_SB_)#ACPI(PCI0)#PCI(0102)#PCI(0000)`

**Reply by k4xx0r** [reply]
After some research I was able to figure out a bit how ASL works. Using [this modified ASL file](https://pastebin.com/b5jBdxrM) , I was able to change the output of [gfxutil](https://github.com/acidanthera/gfxutil/releases) from `pci-bridge@1,2` to `BRG2`.

```
00:00.0 8086:29c0 /PCI0@0/S00@0 = PciRoot(0x1)/Pci(0x0,0x0)
00:01.0 1b36:000c /PCI0@0/S08@1 = PciRoot(0x1)/Pci(0x1,0x0)
00:01.1 1b36:000c /PCI0@0/pci-bridge@1,1 = PciRoot(0x1)/Pci(0x1,0x1)
00:01.2 1b36:000c /PCI0@0/BRG2@1,2 = PciRoot(0x1)/Pci(0x1,0x2)
00:01.3 1b36:000c /PCI0@0/pci-bridge@1,3 = PciRoot(0x1)/Pci(0x1,0x3)
00:01.4 1b36:000c /PCI0@0/pci-bridge@1,4 = PciRoot(0x1)/Pci(0x1,0x4)
01:00.0 1b36:000e /PCI0@0/S08@1/pci-bridge@0 = PciRoot(0x1)/Pci(0x1,0x0)/Pci(0x0,0x0)
00:01.5 1b36:000c /PCI0@0/pci-bridge@1,5 = PciRoot(0x1)/Pci(0x1,0x5)
00:01.6 1b36:000c /PCI0@0/pci-bridge@1,6 = PciRoot(0x1)/Pci(0x1,0x6)
00:01.7 1b36:000c /PCI0@0/pci-bridge@1,7 = PciRoot(0x1)/Pci(0x1,0x7)
04:00.0 1002:694c /PCI0@0/BRG2@1,2/GFX1@0 = PciRoot(0x1)/Pci(0x1,0x2)/Pci(0x0,0x0)
00:07.0 8086:2934 /PCI0@0/UHC1@7 = PciRoot(0x1)/Pci(0x7,0x0)
03:00.0 1af4:1043 /PCI0@0/pci-bridge@1,1/pci1af4,1043@0 = PciRoot(0x1)/Pci(0x1,0x1)/Pci(0x0,0x0)
05:00.0 1002:ab08 /PCI0@0/pci-bridge@1,3/HDEF@0 = PciRoot(0x1)/Pci(0x1,0x3)/Pci(0x0,0x0)
02:01.0 1234:1111 /PCI0@0/S08@1/pci-bridge@0/GFX0@1 = PciRoot(0x1)/Pci(0x1,0x0)/Pci(0x0,0x0)/Pci(0x1,0x0)
00:07.1 8086:2935 /PCI0@0/UHC2@7,1 = PciRoot(0x1)/Pci(0x7,0x1)
02:02.0 15ad:07b0 /PCI0@0/S08@1/pci-bridge@0/ethernet@2 = PciRoot(0x1)/Pci(0x1,0x0)/Pci(0x0,0x0)/Pci(0x2,0x0)
00:07.2 8086:2936 /PCI0@0/UHC3@7,2 = PciRoot(0x1)/Pci(0x7,0x2)
00:07.7 8086:293a /PCI0@0/EH01@7,7 = PciRoot(0x1)/Pci(0x7,0x7)
00:1f.0 8086:2916 /PCI0@0/ISA@1F = PciRoot(0x1)/Pci(0x1F,0x0)
00:1f.3 8086:2930 /PCI0@0/SMB0@1F,3 = PciRoot(0x1)/Pci(0x1F,0x3)
00:1f.2 8086:2922 /PCI0@0/pci8086,2922@1F,2 = PciRoot(0x1)/Pci(0x1F,0x2)
```



Unfortunately, it didn't have any effect on the GPU. I also tried to use the DeviceProperties field in the config.plist, but other than spoofing to a RX570 it wouldn't have any impact on the output. 

It feels like I'm very near to solving this. Do you have any hint where to approach next?
